vllm/distributed/kv_transfer/kv_connector/v1/nixl_connector.py

```diff
@@ -1202,7 +1218,7 @@ def register_kv_caches(self, kv_caches: dict[str, torch.Tensor]):
 
         # TODO (NickLucche): Get kernel_block_size in a cleaner way
         # NHD default "view" for non-MLA cache
-        if self.device_type == "cpu":
+        if self.device_type == "cpu" or self.cross_layers:
             block_size_position = -2
         else:
             block_size_position = -2 if self.use_mla else -3
```

Comment:
As I suggested to eliminate the `self.cross_layers` variable,
I suggest that we move this entire logic behind a new property `TpKVTopology.block_size_position`.
cc @NickLucche